src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/ValidationAttributeStore.cs

```diff
@@ -216,10 +214,7 @@ internal bool TryGetPropertyStoreItem(string propertyName, out PropertyStoreItem
                 {
                     lock (_syncRoot)
                     {
-                        if (_propertyStoreItems == null)
```

Comment:
Why did you delete this check?  Isn't it necessary for thread-safety as part of the double-checked locking scheme bein employed?